src/components/AutoComplete/AutoComplete.tsx

```diff
@@ -0,0 +1,742 @@
+            } else {
+              valueNode.current.set(option.value, option);
+              const title = getTextFromNodes(option.label).toLowerCase();
+              if (title.includes(lowerCasedSearch)) {
+                visibleList.current.push(option.value);
+                if (!disabled) {
+                  navigatable.current.push(option.value);
+                }
+              }
+              return {
+                disabled: option.disabled,
+                value: option.value,
+                title: getTextFromNodes(option.label),
+              };
+            }
+          })
+        );
+      } else if (children) {
+        setList(childrenToComboboxItemArray(children, updateElements));
+      }
+    },
+    [disabled, search, updateElements]
+  );
+
+  useEffect(() => {
+    updateList(children, options);
+  }, [children, options, updateList]);
+
+  const onKeyDown: KeyboardEventHandler<HTMLInputElement> = e => {
+    if (!e.defaultPrevented) {
+      if (e.key === "Enter") {
+        e.preventDefault();
+        if (highlighted) {
+          onSelect(highlighted);
+        }
+      } else if (["ArrowUp", "ArrowDown", "Home", "End"].includes(e.key)) {
+        e.preventDefault();
+        let nextHighlightedValue = highlighted;
+        const highlightedIndex = navigatable.current.findIndex(
+          value => value === highlighted
+        );
+        if (e.key === "ArrowUp") {
+          if (highlightedIndex === 0) {
+            nextHighlightedValue = navigatable.current[navigatable.current.length - 1];
+          } else {
+            nextHighlightedValue = navigatable.current[highlightedIndex - 1];
+          }
+        } else if (e.key === "ArrowDown") {
+          e.preventDefault();
+          if (highlightedIndex === navigatable.current.length - 1) {
+            nextHighlightedValue = navigatable.current[0];
+          } else {
+            nextHighlightedValue = navigatable.current[highlightedIndex + 1];
+          }
+        } else if (e.key === "End") {
+          e.preventDefault();
+          nextHighlightedValue = navigatable.current[navigatable.current.length - 1];
+        } else if (e.key === "Home") {
+          nextHighlightedValue = navigatable.current[0];
+          e.preventDefault();
+        }
+        setHighlighted(nextHighlightedValue);
+      }
+    }
+  };
+  const isHidden = useCallback(
+    (value?: string) => {
+      return !visibleList.current.includes(value ?? "");
+    },
+    [visibleList]
+  );
+
+  const optionContextValue = useMemo(() => {
+    return {
+      search,
+      updateHighlighted: setHighlighted,
+      highlighted,
+      isHidden,
+      onSelect,
+      selectedValue: value,
+    };
+  }, [search, highlighted, isHidden, onSelect, value]);
+
+  const conditionalProps: Partial<SelectOptionProp> = {};
+  if (options) {
+    conditionalProps.options = options;
+  } else {
+    conditionalProps.children = children;
+  }
+
+  return (
+    <FormRoot
```

Comment:
i feel we should reuse our search field component and attach the autocomplete behavior to it. 
Is there a reason not to reuse the search component ?

Reply:
The search Field is being used
The outer layer is different because we only require the styling and not the input element
Thats why I added it like that
The search field would be visible only when the element is clicked and the popover is open

Reply:
Also in case of error the error clicking should not open the popover thats why we can't reuse it

Reply:
this can be simplified to 

```js
   <Trigger
    data-testid="autocomplete-trigger"
    asChild
  >
    <div>
      <SearchField
        value={search}
        onChange={onUpdateSearch}
        placeholder={placeholder}
        onClick={() => setOpen(true)}
      />
    </div>
  </Trigger>
```
and then in the <PopoverContent>
        
```js
onOpenAutoFocus={event => {
        event.preventDefault();}
}
```

in this way we can avoid recreating the entire style of the SearchField and we can remove the FormRoot, FormElementContainer etc